src/InsecureCredentialsWrapper.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/*
+ * Copyright 2024 Google LLC
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace Google\ApiCore;
+
+/**
+ * For connect to emulator.
+ */
+class InsecureCredentialsWrapper extends CredentialsWrapper
+{
+    public function __construct()
+    {
+    }
+
+    public function getAuthorizationHeaderCallback($audience = null)
+    {
+        return null;
```

Comment:
I have given a hint on it can return null :P

Reply:
Oh okay, I didn't know this is an issue, I will change it to empty array then, Thank you for finding this.

Reply:
But we need a callable right?

Reply:
If you look at my suggestion, it does return a callable! but yes, what you have is even better (more concise 😄 )

Reply:
Oops, I misunderstood the suggestion :-/, Yeah, I initially read it as returning an array, I didn't notice it was actually the return statement itself. Thank you for the great suggestion and sorry about the misunderstanding!

Reply:
We should also create tests for this. I made one in order to see if the HTTP transport was causing a PHP fatal error when using the `InsecureCredentialsWrapper` (it was!)

https://github.com/googleapis/gax-php/pull/539

It would be good to add a test for gRPC as well (see https://github.com/googleapis/google-cloud-php/pull/7081). We had this class break recently as a result of it being in Core rather than in GAX, so I'm happy to see us moving it here to prevent this from happening in the future.

Reply:
Thank you for the test PR, Added a new test for gRPC.

Reply:
We need grpc extension for this to work in the PR testing env!

Reply:
The tests are failing with the following error:
`Class "Grpc\ChannelCredentials" not found`

But they are even used in the gax test itself . see https://github.com/googleapis/gax-php/blob/9b128406e64bebe2718872966c73fcfb8a53a417/src/Testing/MockGrpcTransport.php#L36C10-L36C28

Reply:
We need to check the Grpc extension loaded or not, if not, skip it. Locally ran this test and running perfectly.

> I made one in order to see if the HTTP transport was causing a PHP fatal error when using the InsecureCredentialsWrapper (it was!)

Thank you for catching the failure via tests, it is helpful 👍